### PR TITLE
Add key to signer

### PR DESCRIPTION
**Pull request by aaasen**
- Add key ID to signer.
 - Remove `KeyStoreTODO`.
 - Add `MemoryKeyStore`.
 - Replace `KeyStore.NewKey` with `KeyStore.WriteKey`.
 - Add utility to generate new keys.

### CODE REVIEW

File: jwt.go [hunk 1]
```diff
@@ -32,13 +32,15 @@ func (c Claims) Valid() error {
...
 // Signer is a signing authority for creating authentication tokens.
 type Signer struct {
+	keyID    string
 	keyStore keystore.KeyStore
 	keyFunc  jwt.Keyfunc
 }
 
 // NewSigner creates a token factory given a persistent key store.
-func NewSigner(ks keystore.KeyStore) *Signer {
+func NewSigner(keyID string, ks keystore.KeyStore) *Signer {
```
Comment (ckarenz) [inline jwt.go]: This param might be more appropriate on NewToken.

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,6 @@
 # Generated files
 mock_*
 mocks/
+
+# Binaries
+generate-key
```
Comment (ckarenz) [inline .gitignore]: I don't see the source for this.

Comment (aaasen) [reply .gitignore]: It is from `go build ./cmd/generate-key`

Comment (aaasen) [reply .gitignore]: Ah, forgot to add it.

File: jwt.go (hunk 1 above)
Comment (aaasen) [reply jwt.go]: Done.

File: cmd/generate-key/main.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	}
+
+	key, err := randomHex(32)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("ID:  %s\nKey: %s\n", id, key)
+}
+
+func randomHex(n int) (string, error) {
```
Comment (ckarenz) [inline cmd/generate-key/main.go]: Why remove the test?  I'd imagine we still want to validate lengths where human eyeballs may fail.

**Pull request merged**